docs/design/storage-auto-scale.md

```diff
@@ -72,36 +72,54 @@ Controller:
+4) If the expansion is in progress, check the progress and then requeue each 1 minute until the expansion is completed successfully(jump to step 13).
    
-4) If the LSO storageclass is detected in the storageClassDeviceSet, raise a warning and do not recocnile further.
+5) If no-expansion is in progress,
    
-5) If the highest osd percentage reported in the sync map is more than osdScalingThresholdPercent(70%) means reaching osd nearfull, scaling is needed.
+   1) Check the (status.phase == "InProgress"), if yes move the status phase to succeeded.
 
-6) If scaling is needed calculate the `expectedOsdSize` and `expectedOsdCount`.
+   2) Proceed with further steps.
+
+6) If the LSO storageclass is detected in the storageClassDeviceSet, raise a warning and do not recocnile further.
+   
+7) If the highest osd percentage reported in the sync map is more than osdScalingThresholdPercent(70%) means reaching osd nearfull, scaling is needed.
+
+8) If scaling is needed calculate the `expectedOsdSize` and `expectedOsdCount`.
        
     1)  If the Osd size is less than maxOsdSize(default:8Tib), do vertical scaling by doubling the each osd sizes for that device class.
             
     2)  If the Osd size is equal to maxOsdSize(default:8Tib), do a horizontal scaling, by adding 1 osd of maxOsdSize(default:8Tib) on each `storageDeviceSet`.
    
-7) Calculate the `expectedStorageCapacity` based on expected size and count.
+9)  Calculate the `expectedStorageCapacity` based on expected size and count.
    
-8) Check if the `storageCapacityLimit` > `expectedStorageCapacity`.
+10) Check if the `storageCapacityLimit` > `expectedStorageCapacity`.
         
     1) If yes, Update `phase` to `InProgress`  and `lastExpansionStartTime` as `current-time` on the `StorageAutoScaling` CR which need scaling.
         
     2) If no, don't reconcile further and the set the `storageCapacityLimitReached` as true in the status.
 
-9)  Update the status, set the `expectedOsdSize` and `expectedOsdCount` to reflect the new expected value and also set `startOsdSize` and `startOsdCount` with current storagecluster values.
+11) Update the status, set the `expectedOsdSize` and `expectedOsdCount` to reflect the new expected value and also set `startOsdSize` and `startOsdCount` with current storagecluster values.
    
-10) Scale by patching the Storagecluster, with all the device sets update needed at the same time.
+12) Scale by patching the Storagecluster, with all the device sets update needed at the same time.
 
-11) Verify and Alert:
+13) Verify and Alert:
    
     1) Verify the Storagecluster whether the new osds are added or scaled in size, for all the device sets.
     
+       1) For vertical scaling, Query osd size from Prometheus and match it with storagecluster.spec..size.
```

Comment:
@travisn should we requery prometheues for osd size and count during verify?
Or we should query at the start only

Reply:
If we just triggered the autoscale, I expect it will take at least a few minutes for the OSDs to scale up/out. So I would just expect we can requeue immediately here instead of checking for completion.